feature/settings/src/main/java/com/google/jetpackcamera/settings/ui/SettingsComponents.kt

```diff
@@ -253,6 +256,69 @@ fun CaptureModeSetting(currentCaptureMode: CaptureMode, setCaptureMode: (Capture
     )
 }
 
+@Composable
+fun TargetFpsSetting(
+    modifier: Modifier = Modifier,
+    currentTargetFps: TargetFrameRate,
+    supportedFps: List<Int>,
+    setTargetFps: (TargetFrameRate) -> Unit
+) {
+    BasicPopupSetting(
+        modifier = modifier,
+        title = stringResource(id = R.string.fps_title),
+        enabled = supportedFps.isNotEmpty(),
+        leadingIcon = null,
+        description = if (supportedFps.isEmpty()) {
+            stringResource(id = R.string.fps_description_unavailable)
+        } else {
+            when (currentTargetFps) {
+                TargetFrameRate.TARGET_FPS_NONE -> stringResource(
+                    id = R.string.fps_description_none
+                )
+                TargetFrameRate.TARGET_FPS_15 -> stringResource(id = R.string.fps_description_15)
+                TargetFrameRate.TARGET_FPS_30 -> stringResource(id = R.string.fps_description_30)
+                TargetFrameRate.TARGET_FPS_60 -> stringResource(id = R.string.fps_description_60)
+            }
+        },
+        popupContents = {
+            Column(Modifier.selectableGroup()) {
+                Text(
+                    text = stringResource(id = R.string.fps_stabilization_disclaimer),
+                    fontStyle = FontStyle.Italic,
+                    color = MaterialTheme.colorScheme.onPrimaryContainer
+                )
+
+                SingleChoiceSelector(
+                    text = stringResource(id = R.string.fps_selector_none),
+                    selected = currentTargetFps == TargetFrameRate.TARGET_FPS_NONE,
+                    onClick = { setTargetFps(TargetFrameRate.TARGET_FPS_NONE) }
+                )
+
+                SingleChoiceSelector(
+                    text = stringResource(id = R.string.fps_selector_15),
+                    selected = currentTargetFps == TargetFrameRate.TARGET_FPS_15,
+                    onClick = { setTargetFps(TargetFrameRate.TARGET_FPS_15) },
+                    enabled = supportedFps.contains(15)
+                )
+
+                SingleChoiceSelector(
+                    text = stringResource(id = R.string.fps_selector_30),
+                    selected = currentTargetFps == TargetFrameRate.TARGET_FPS_30,
+                    onClick = { setTargetFps(TargetFrameRate.TARGET_FPS_30) },
+                    enabled = supportedFps.contains(30)
+                )
+
+                SingleChoiceSelector(
+                    text = stringResource(id = R.string.fps_selector_60),
+                    selected = currentTargetFps == TargetFrameRate.TARGET_FPS_60,
+                    onClick = { setTargetFps(TargetFrameRate.TARGET_FPS_60) },
+                    enabled = supportedFps.contains(60)
+                )
```

Comment:
You can just have a loop here to add the values from supportedFps:

```
supportedFps.forEach { fpsOption ->
  SingleChoiceSelector(
     text = "%d".format(fpsOption), // Should localize automatically
     selected = currentTargetFps == fpsOption,
     onClick = { setTargetFps(fpsOption) }
     // enabled = supportedFps.contains(30) <--- not needed since the option won't exist if its not supported
  )
```

Reply:
Thanks! but i think i'd still like to keep the disabled out fps options. It follows the pattern of the other settings having greyed out options and I think having a visible indicator is nice.
I think its nice to have a visible indication of "we checked for this and its unavailable"
Instead of only telling "this is what is available to you"

Reply:
SGTM. This is probably better for now anyways since it will make us have to think harder about the layout if we add many more choices.